### PR TITLE
Enh color names

### PR DESCRIPTION
**Pull request by tacaswell**
This supersedes https://github.com/matplotlib/matplotlib/pull/7343/files

This is the result of an email conversation with Maureen Stone from Tableau.

We still need a link target for the default style changes to tableau.

The naming prefix 'T10' might need a bit more thought as these colors are actually from an older version of Tableau.  They have since updated their colors and we should not name our selves into a corner.

@trpham would you be offended if I squashed your commits into 1?

### CODE REVIEW

**Comment by QuLogic** [comment]
Still hoping for a 'grey' variant; not sure what to do about the prefix.

File: doc/users/dflt_style_changes.rst [hunk 1]
```diff
@@ -32,8 +32,13 @@ Colors in default property cycle
...
 
 The colors in the default property cycle have been changed from
-``['b', 'g', 'r', 'c', 'm', 'y', 'k']`` to the `Vega category10 palette
-<https://github.com/vega/vega/wiki/Scales#scale-range-literals>`__
+``['b', 'g', 'r', 'c', 'm', 'y', 'k']`` to the ``__ to the category10
+color palete used by `Vega
+<https://github.com/vega/vega/wiki/Scales#scale-range-literals>`__ and
+`d3
+<https://github.com/d3/d3-3.x-api-reference/blob/master/Ordinal-Scales.md#category10>`__
+originally developed at Tableau.
```
Comment (efiring) [inline doc/users/dflt_style_changes.rst]: Add link here to the Tableau blog post describing their development?

Comment (tacaswell) [inline doc/users/dflt_style_changes.rst]: I offered to let Tableau pick where this link goes, still waiting to hear back from them.

**Comment by NelleV** [comment]
If we go with T10, it implies that we will either have another namespace for the vega 20 colorscheme, or that we won't add those colors.
I'd rather go with the prefix `tableau`.

**Comment by NelleV** [comment]
This is set to the milestone 2.0. Is that on purpose?
If so, do you mind if I also work on this to get this out of the door ASAP?

**Comment by tacaswell** [comment]
Yes, milestone is intentional.

I would have claimed this is low risk, but I seem to have broken everything...

**Comment by NelleV** [comment]
It is because the color name cannot contain capital letters with our current code.

**Comment by NelleV** [comment]
(culprit being colors.py l.156, that checks that the *lower case* of the colorname provided is in a dictionnary).

**Comment by mcstone** [comment]
Just confirming that we at Tableau are quite happy to have you folks include any of our colors as long as they are properly credited (using tableau as the prefix would make that clear, of course, though it is a bit long). The Classic Tableau 10 and 20 have been in use for a while, are the ones currently under discussion. But, I have a new set of default colors that got designed according to this [article](https://www.tableau.com/about/blog/2016/7/colors-upgrade-tableau-10-56782), as well as many more palettes.

**Comment by NelleV** [comment]
@tacaswell should we change the prefix to tableau ? it is longer to type, but I think it is worth having an explicit name (T10 is a bit obscure). It also allows us to have only one namespace for the 10 and 20 color schemes and in addition fixes the capitalization issue.

**Comment by mcstone** [comment]
Another prefix to consider is tc (for Tableau Color). Putting the explanation in a comment next to the definition would both provide the mnemonic and give credit.

**Comment by NelleV** [comment]
I still prefer the explicit name "tableau". It is longer to type, but easier to remember.
(I think I am going to loose this battle :p)

File: doc/users/dflt_style_changes.rst [hunk 2]
```diff
@@ -32,8 +32,13 @@ Colors in default property cycle
 --------------------------------
 
 The colors in the default property cycle have been changed from
-``['b', 'g', 'r', 'c', 'm', 'y', 'k']`` to the `Vega category10 palette
-<https://github.com/vega/vega/wiki/Scales#scale-range-literals>`__
+``['b', 'g', 'r', 'c', 'm', 'y', 'k']`` to the ``__ to the category10
```
Comment (NelleV) [inline doc/users/dflt_style_changes.rst]: I think there is a problem here.

Comment (tacaswell) [inline doc/users/dflt_style_changes.rst]: Thanks.  I swear I know how to delete text....

File: doc/users/colors.rst [hunk 3]
```diff
@@ -17,6 +17,10 @@ it can be provided as:
 * a name from the `xkcd color survey <https://xkcd.com/color/rgb/>`__
   prefixed with ``'xkcd:'`` (e.g., ``'xkcd:sky blue'``)
 * one of ``{'C0', 'C1', 'C2', 'C3', 'C4', 'C5', 'C6', 'C7', 'C8', 'C9'}``
+* one of ``{'tc:blue', 'tc:orange', 'tc:green',
+  'tc:red', 'tc:purple', 'tc:brown', 'tc:pink',
+  'tc:gray', 'tc:olive', 'tc:cyan'}`` which are the Tableau Colors from the
+  'T10' categorical palate (which is the default color cycle).
```
Comment (QuLogic) [inline doc/users/colors.rst]: palette

File: doc/users/dflt_style_changes.rst [hunk 4]
```diff
@@ -32,8 +32,13 @@ Colors in default property cycle
 --------------------------------
 
 The colors in the default property cycle have been changed from
-``['b', 'g', 'r', 'c', 'm', 'y', 'k']`` to the `Vega category10 palette
-<https://github.com/vega/vega/wiki/Scales#scale-range-literals>`__
+``['b', 'g', 'r', 'c', 'm', 'y', 'k']`` to the category10
+color palete used by `Vega
```
Comment (QuLogic) [inline doc/users/dflt_style_changes.rst]: palette

**Comment by stonebig** [comment]
is this the last thing blocking Matplotlib-2.0 ? could it be postpône ?

**Comment by tacaswell** [comment]
color names are part of the API.

**Comment by dopplershift** [comment]
👍  to `tc` prefix--tableau is a nightmare for spelling with our predominantly english-speaking audience.

**Comment by anntzer** [comment]
+1 for tc as well.

**Comment by tacaswell** [comment]
@phobson the whole coverage report looks off, it says this PR adds 65 files.

**Comment by NelleV** [comment]
I still don't like "tc" as a prefix. It is very obscure. Someone suggested to me "tab", as it is both short, and avoids the "eau" that english speakers struggle with remembering.

Are we interested in exploring other possible prefixes?

**Comment by NelleV** [comment]
(Else this looks ready to be merged)

**Comment by dopplershift** [comment]
"tab" works for me.

**Comment by phobson** [comment]
@NelleV I think this ship has sailed :/. 

My personal preference would be `tableau`, but given that the palette's designer has blessed `tc`, that's probably the best option.

**Comment by anntzer** [comment]
@QuLogic @tacaswell Isn't this just a matter of adding it in the T10_COLORS dictionary?

**Comment by tacaswell** [comment]
I am wary of adding extra colors given that we are adopting an existing palette.

We already have `'grey'` and `'gray'` the un-prefixed colors and floats as strings get mapped to shades of gray.

```python
import matplotlib.colors as mc
print(mc.to_hex(mc.to_rgb('gray'))) 
# '#808080'
```

**Comment by QuLogic** [comment]
They are, as with the unprefixed colours, expected to be the same colour, not some new value.

**Comment by QuLogic** [comment]
And in the unprefixed set, we have darkgray/darkgrey, darkslategray/darkslategrey, dimgray/dimgrey, gray/grey, lightgray/lightgrey, lightslategray/lightslategrey, slategray/slategrey with every pair equal to each other, so I don't see why this new palette can't have the same aliasing.

Interestingly, the xkcd palette only spells it 'grey'.

**Comment by tacaswell** [comment]
@QuLogic Sorry, completely missed that we already had 'tc:gray' and was responding to something un-related to what you were asking :sheep: 

My only concern about including two spellings of 'gray' in the palette is that users may want to use `list(matplotlib.colors.T10_COLORS.items())` as a color cycle.  Of the dictionaries we currently have, it is the only one small enough to usable for that, so this a completely made up use case, but one I could see growing in the future.

**Comment by anntzer** [comment]
That would require (in practice) the dict being an OrderedDict, which may not be a bad idea in itself...

**Comment by tacaswell** [comment]
@anntzer That is also a good point and idea!

**Comment by anntzer** [comment]
(but then you need to know where to put the other spelling of grey/gray in the ordereddict)

**Comment by tacaswell** [comment]
Looking at the work you did in `colors.py` with the `_colors_full_mapping` we can do the gray/grey normalization there and leave `TABLEAU_COLORS` and `XKCD_COLORS` spelled as the source material spells them.  I am not 100% that the spelling it tableau is 'cannonical', might want to normalize it to 'grey' to match xkcd (which makes the duplication code in colors.py a bit more uniform).

**Comment by tacaswell** [comment]
To try and summarize all of the prefix discussions:

 - tableau
   - pro: very explicit
   - con: long, any future varients would be even longer
   - con: anglophiles have a hard time spelling it
 - tc
   - pro: short
   - pro: suggested by @mcstone from tableau
   - con: some what obscure to source of naming
   - con: 'c' is redundant as we all know we are talking about colors
   - con(?): my initials
 - t
   - same as tc, but even shorter!
 - tab
   - pro: short
   - pro: much less obscure than 'tc'
   - con: collision with head namespace of the tab key? (that is the best I can come up with)


and I think I have written my self into preferring 'tab'

**Comment by dopplershift** [comment]
Because of the initials aspect, at the very minimum we need a comment saying explicitly what `tc` stands for.

**Comment by efiring** [comment]
On 2017/01/16 8:03 AM, Thomas A Caswell wrote:
> and I think I have written my self into preferring 'tab'

"tab" is fine with me.

**Comment by NelleV** [comment]
LGTM :+1:

**Comment by tacaswell** [comment]
backported to v2.x as 4e1df0b48

**Pull request merged**